Code/GraphMol/RGroupDecomposition/RGroupScore.h

```diff
@@ -40,6 +42,14 @@ struct CartesianProduct {
     return increment(0);
   }
 
+  size_t value() {
```

Comment:
This needs an explanation. I think I know what it's doing, but I'm not really sure why it's being done.

Reply:
Unless this is the "stable sort"

Reply:
Yes, it is the last resort criterion. If nothing else broker the tie, we look at the permutation value converted to integer. See explanation above.